### ISSUE TITLE
tutorial lesson_12_using_the_gpu errors on macOS

### ISSUE THREAD

**Issue by mikewoodworth** [halide/Halide#4606]
trying to invoke GPU tutorial on macOS Catalina, errors with the following output:

Running pipeline on CPU:
Running pipeline on GPU:
Target: x86-64-osx-avx-avx2-f16c-fma-metal-sse41
Testing GPU correctness:
Error: Metal: cannot allocate system default device.
Abort trap: 6

**Comment by shoaibkamil** [comment]
On master, I see:
```
$ HL_JIT_TARGET=host-metal ./lesson_12
Running pipeline on CPU:
Running pipeline on GPU:
Requested GPU(s) are not supported. (Do you have the proper hardware and/or driver installed?)
No GPU target available on the host
Testing performance on CPU:
6.6550 milliseconds
```

It's not a crash, but it doesn't work.  I tried compiling with `-framework Metal` but that didn't seem to change anything.

**Comment by mikewoodworth** [comment]
I haven't tried pre catalina yet, will do so just to confirm it isnt a security thing

**Comment by shoaibkamil** [comment]
Also:
```
$  HL_JIT_TARGET=host-metal-debug make correctness_hello_gpu
Defining function...
Realizing function...
Entering Pipeline f
Target: x86-64-osx-avx-avx2-debug-f16c-fma-jit-metal-sse41-user_context
 Input (void *) __user_context: 0x7ffeea224e50
 Output Buffer f: buffer(0, 0x0, 0x0, 0, float32, {0, 32, 1}, {0, 32, 32})
Metal - Allocating: MTLCreateSystemDefaultDevice
Error: (at test/correctness/hello_gpu.cpp:23)
Metal: cannot allocate system default device.
bash: line 1: 38344 Abort trap: 6           /Users/kamil/Documents/work/adobe/halide_master/bin/correctness_hello_gpu
make: *** [correctness_hello_gpu] Error 134
```

Seems like something is broken on Catalina for Metal via JIT.

**Comment by steven-johnson** [comment]
Could be an injection from https://github.com/halide/Halide/pull/4588 -- I don't get this locally on my Mac, but I'm not running Catalina.

**Comment by shoaibkamil** [comment]
Using LLDB, I see this:

```
Process 23235 launched: '/Users/kamil/Documents/work/adobe/halide_master/bin/correctness_hello_gpu' (x86_64)
warning: unable to find and load segment named '__OBJC_RO' at 0x7fff780ec000 in '/usr/lib/libobjc.A.dylib' in macosx dynamic loader plug-in.
warning: unable to find and load segment named '__OBJC_RW' at 0x7fff9854f080 in '/usr/lib/libobjc.A.dylib' in macosx dynamic loader plug-in.
Defining function...
Realizing function...
Entering Pipeline f
Target: x86-64-osx-avx-avx2-debug-f16c-fma-jit-metal-sse41-user_context
 Input (void *) __user_context: 0x7ffeefbfef10
 Output Buffer f: buffer(0, 0x0, 0x0, 0, float32, {0, 32, 1}, {0, 32, 32})
Metal - Allocating: MTLCreateSystemDefaultDevice
2020-03-03 10:18:46.766417-0500 correctness_hello_gpu[23235:23736025] Use of MTLCreateSystemDefaultDevice is not supported for non-interactive (commandline or daemon) apps. Use MTLCopyAllDevices(WithObserver) instead.
Error:
Metal: cannot allocate system default device.
Process 23235 stopped
```

Patch incoming.

**Comment by shoaibkamil** [comment]
Fixed by #4668.
